### PR TITLE
fix: enable python3 %check tests

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Run python3 regression test suite during the %check section.

Signed-off-by: Chris Co <chrco@microsoft.com>

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: [532346](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=532346&view=results)

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM

**Comment by mfrw** [comment]
for some reason the arm build is still going on

**Comment by christopherco** [comment]
> for some reason the arm build is still going on

It looks like there was an error failing to initilize the ccache manager

```
time="2024-03-21T16:33:02Z" level=info msg="* Creating a ccache manager instance *"
time="2024-03-21T16:33:02Z" level=info msg="  ccache root folder         : ()"
time="2024-03-21T16:33:02Z" level=info msg="  ccache remote configuration: ()"
time="2024-03-21T16:33:02Z" level=warning msg="Failed to initialize the ccache manager:\nCCache root directory cannot be empty."
time="2024-03-21T16:33:02Z" level=info msg="Building (python3-3.12.0-3.azl3.src.rpm)."
time="2024-03-21T16:33:02Z" level=error msg="Failed to build SRPM '/mnt/vss/_work/1/s/build/INTERMEDIATE_SRPMS/python3-3.12.0-3.azl3.src.rpm'. For details see log file: /mnt/vss/_work/1/s/build/logs/pkggen/rpmbuilding/python3-3.12.0-3.azl3.src.rpm.test.log ."
time="2024-03-21T16:33:02Z" level=panic msg="chroot directory (/mnt/vss/_work/1/s/worker/chroot/python3-3.12.0-3.azl3_TEST_BUILD) already exists"
```
Not sure why this happened.

Requeued version passes - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533051&view=results

**Review by mfrw — APPROVED** [review]
LGTM

**Review by dmcilvaney — APPROVED** [review]

File: SPECS/python3/python3.spec [hunk 1]
```diff
@@ -168,8 +168,8 @@ find %{buildroot}%{_libdir} -name '*.o' -delete
 rm %{buildroot}%{_bindir}/2to3
 rm -rf %{buildroot}%{_bindir}/__pycache__
 
-# %check
-# make  %{?_smp_mflags} test
+%check
+%{buildroot}%{_bindir}/python3 -m test
```
Comment (PawelWMS) [inline SPECS/python3/python3.spec:172]: Does this run all the tests? Or would `-m test.regrtest` or `-m test.pythoninfo` be separate sets of tests?

**Review by anphel31 — APPROVED** [review]

File: SPECS/python3/python3.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/python3/python3.spec:172]: This runs all the tests

Comment (christopherco) [reply SPECS/python3/python3.spec:172]: Some additional documentation on the test package - https://docs.python.org/3/library/test.html#module-test.regrtest

> The [test](https://docs.python.org/3/library/test.html#module-test) package can be run as a script to drive Python’s regression test suite, thanks to the [-m](https://docs.python.org/3/using/cmdline.html#cmdoption-m) option: python -m test. Under the hood, it uses [test.regrtest](https://docs.python.org/3/library/test.html#module-test.regrtest); the call python -m test.regrtest used in previous Python versions still works. Running the script by itself automatically starts running all regression tests in the [test](https://docs.python.org/3/library/test.html#module-test) package. It does this by finding all modules in the package whose name starts with test_, importing them, and executing the function test_main() if present or loading the tests via unittest.TestLoader.loadTestsFromModule if test_main does not exist. The names of tests to execute may also be passed to the script. Specifying a single regression test (python -m test test_spam) will minimize output and only print whether the test passed or failed.

Comment (PawelWMS) [reply SPECS/python3/python3.spec:172]: Yes, but are we also running the two mentioned? I see that Fedora runs them separately, so I'm curious, if we can test even more.

Comment (christopherco) [reply SPECS/python3/python3.spec:172]: I dove into one of fedora's python3.12.0 builds in koji ([https:/kojipkgs.fedoraproject.org//packages/python3.12/3.12.0/1.fc40/data/logs/x86_64/build.log](https://kojipkgs.fedoraproject.org//packages/python3.12/3.12.0/1.fc40/data/logs/x86_64/build.log)):
- `test.pythoninfo` prints useful debugging information, which would be nice to add. I'll look at adding this in another PR.
- `test.regrtest` runs what appears to be the same tests, though there is a slight difference. In Fedora's build, it is running 483 tests, but in ours we're running 463. I'll see if I can compare the two to figure out the 20 test differences. A random sampling of tests between ours and Fedora's showed that the test scenarios were common

**Pull request merged**